packages/salesforcedx-vscode-core/src/conflict/conflictDetectionService.ts

```diff
@@ -0,0 +1,218 @@
+      .withLogName('conflict_detect_retrieve_org_source')
+      .build();
+  }
+
+  public buildMetadataApiConvertOrgSourceCommand(
+    rootDir: string,
+    outputDir: string
+  ): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('conflict_detect_convert_org_source'))
+      .withArg('force:mdapi:convert')
+      .withFlag('--rootdir', rootDir)
+      .withFlag('--outputdir', outputDir)
+      .withLogName('conflict_detect_convert_org_source')
+      .build();
+  }
+
+  public async checkForConflicts(
+    data: ConflictDetectionConfig
+  ): Promise<DirectoryDiffResults> {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const results = await this.checkForConflictsInternal(
+      getRootWorkspacePath(),
+      data,
+      cancellationTokenSource,
+      cancellationToken
+    );
+    return results;
+  }
+
+  private async checkForConflictsInternal(
+    projectPath: string,
+    data: ConflictDetectionConfig,
+    cancellationTokenSource: any,
+    cancellationToken: any
+  ): Promise<DirectoryDiffResults> {
+    const tempMetadataPath = this.getCachePath(data.usernameOrAlias);
+
+    // 1: create the shadow directory
+    const manifestPath = path.join(tempMetadataPath, 'package.xml');
+    try {
+      shell.mkdir('-p', tempMetadataPath);
+      shell.cp(data.manifest, manifestPath);
+    } catch (error) {
+      this.reportError('error_creating_packagexml', error);
+      return Promise.reject();
+    }
+
+    // 2: retrieve unmanaged org source to the shadow directory
+    await this.executeCommand(
+      this.buildRetrieveOrgSourceCommand(
+        data.usernameOrAlias,
+        tempMetadataPath,
+        manifestPath
+      ),
+      projectPath,
+      cancellationTokenSource,
+      cancellationToken
+    );
+
+    // 3: unzip retrieved source
+    const unpackagedZipFile = path.join(tempMetadataPath, 'unpackaged.zip');
+    try {
+      const zip = new AdmZip(unpackagedZipFile);
+      zip.extractAllTo(tempMetadataPath, true);
+    } catch (error) {
+      this.reportError('error_extracting_org_source', error);
+      return Promise.reject();
+    }
+
+    // 4: convert org source to decomposed (source) format
+    const unconvertedSourcePath = path.join(tempMetadataPath, 'unpackaged');
+    const convertedSourcePath = path.join(tempMetadataPath, 'converted');
+    await this.executeCommand(
+      this.buildMetadataApiConvertOrgSourceCommand(
+        unconvertedSourcePath,
+        convertedSourcePath
+      ),
+      projectPath,
+      cancellationTokenSource,
+      cancellationToken
+    );
+
+    // 5: diff project directory (local) and retrieved directory (remote)
+    // Assume there are consistent subdirs from each root i.e. 'main/default'
+    const localSourcePath: string = path.join(projectPath, data.packageDir);
+    const diffs = this.differ.diff(localSourcePath, convertedSourcePath);
```

Comment:
It would be useful to log how many files where diffed and how many of those had conflicts.

Reply:
That information is now available from the differ, I can log that in the postconditionchecker, where I log the filenames.

Reply:
@lcampos Are you suggesting logging for the users benefit, or using telemetry for metrics?

Reply:
For both, it'd be useful to understand how many files are diffed and how many of those have conflicts. From a user perspective you'd want to know these since the list of file we're currently printing can get to long. From our perspective this will help us tune the feature to make it more performant, etc.

Reply:
Yes, good idea, I've added the telemetry as well.